src/index/index_factory.cc

```diff
@@ -87,31 +96,54 @@ IndexFactory::MapInstance() {
     static FuncMap func_map;
     return func_map;
 }
+
+IndexFactory::FeatureMap&
+IndexFactory::FeatureMapInstance() {
+    static FeatureMap featureMap;
+    return featureMap;
+}
+
 IndexFactory::GlobalIndexTable&
 IndexFactory::StaticIndexTableInstance() {
     static GlobalIndexTable static_index_table;
     return static_index_table;
 }
 
+bool
+IndexFactory::FeatureCheck(const std::string& name, uint64_t feature) const {
```

Comment:
maybe, return `std::optional<bool>`, which returns `nullopt` if `name` is not known?